packages/ui/src/contexts/SwapProvider/useEvmSwap.tsx

```diff
@@ -484,6 +526,77 @@ export const useEvmSwap: SwapAdapter<OptimalRate> = (
     ],
   );
 
+  const performWrapOperation = useCallback(
+    async (params: SwapParams<WrapOperation> | SwapParams<UnwrapOperation>) => {
+      assertPresent(network, CommonError.NoActiveNetwork);
+      assertPresent(account, CommonError.NoActiveAccount);
+      assertPresent(rpcProvider, CommonError.Unknown);
+      assert(!network.isTestnet, CommonError.UnknownNetwork);
+
+      const { srcToken, destToken, srcDecimals, destDecimals } = params;
+      const userAddress = account.addressC;
+      const tokenAddress =
+        params.quote.type === 'WRAP'
+          ? params.quote.target
+          : params.quote.source;
+      const amount = params.quote.amount;
+
+      const abi = tokenAddress === WETH_ADDRESS ? WETH_ABI : WAVAX_ABI;
+      const buildTxParams = {
+        userAddress,
+        tokenAddress,
+        amount,
+        provider: rpcProvider,
+        abi,
+      };
+      const tx =
+        params.quote.type === 'WRAP'
+          ? await buildWrapTx(buildTxParams)
+          : await buildUnwrapTx(buildTxParams);
+
+      const [txHash, signError] = await resolve(
+        request({
+          method: RpcMethod.ETH_SEND_TRANSACTION,
+          params: [tx],
+        }),
+      );
+
+      if (isUserRejectionError(signError)) {
+        throw signError;
+      } else if (signError || !txHash) {
+        throw swapError(CommonError.UnableToSign, signError);
+      }
+
+      const pendingToastId = showPendingToast();
+
+      rpcProvider.waitForTransaction(txHash).then((receipt) => {
+        const isSuccessful = Boolean(receipt && receipt.status === 1);
```

Comment:
Please consider to use optional chaining
```suggestion
        const isSuccessful = Boolean(receipt?.status === 1);
```

Reply:
23aa31cd6